HealthcareSystem/PatientWebApp/wwwroot/js/appointment_recommendation.js

```diff
@@ -0,0 +1,118 @@
+﻿$(document).ready(function () {
+
+	var newAppointments = [];
+
+	var dtToday = new Date();
+
+	var month = dtToday.getMonth() + 1;
+	var day = dtToday.getDate() + 1;
+	var year = dtToday.getFullYear();
+	if (month < 10)
+		month = '0' + month.toString();
+	if (day < 10)
+		day = '0' + day.toString();
+
+	var minDate = year + '-' + month + '-' + day;
+
+	$('#dateOfExam').attr('min', minDate);
+
+	$.ajax({
+		url: "/api/doctor/all-specialty",
+		type: "GET",
+		dataType: 'json',
+		processData: false,
+		contentType: false,
+		success: function (specialties) {
+			console.log('success - loading specialties from database');
+			if (specialties.length == 0) {
+				console.log("there are no specialties");
+			}
+			else {
+				for (let specialty of specialties) {
+					let specialtyOption = $('<option value="' + specialty.id + '">' + specialty.name + '</option>');
+					$('select#specialties').append(specialtyOption);
+				}
+				let select_specialty = $('#specialties option:selected').val();
+
+				$.ajax({
+					url: '/api/doctor/doctor-specialty/' + select_specialty,
+					type: 'GET',
+					dataType: 'json',
+					processData: false,
+					contentType: false,
+					success: function (doctorsSpecialtes) {
+						for (let doctorSpecialist of doctorsSpecialtes) {
+							let doctorJmbg = doctorSpecialist.doctorJmbg;
+
+							$.ajax({
+								url: '/api/doctor/' + doctorJmbg,
+								type: 'GET',
+								dataType: 'json',
+								processData: false,
+								contentType: false,
+								success: function (doctor) {
+									let doctorName = $('<option value="' + doctor.jmbg + '">' + doctor.name + ' ' + doctor.surname + '</option>');
+									$('#doctors').append(doctorName);
+								},
+								error: function () {
+									console.log("Error getting doctor")
+								}
+							});
+						}
+					},
```

Comment:
The design of your endpoints seems kind of questionable. Why would you do one request to fetch relationships, and then go through them and fetch the entities, instead of just connecting those on the backend in one request? This seems like something that should be down in pretty low levels of the stack, by a join in the database or something, in the service at the highest. Definitely not patched together by the frontend. 
Plus, there is literally a GetDoctorsBySpecialty method in the DoctorRepository.